contracts/cw-staking-denom-voting/src/contract.rs

```diff
@@ -0,0 +1,119 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{to_binary, Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
+use cw2::set_contract_version;
+use cw_core_interface::voting::{TotalPowerAtHeightResponse, VotingPowerAtHeightResponse};
+
+use crate::error::ContractError;
+use crate::msg::{ExecuteMsg, InstantiateMsg, MigrateMsg, QueryMsg};
+use crate::state::{DAO, STAKING_MODULE};
+
+const CONTRACT_NAME: &str = "crates.io:cw-staking-denom-voting";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    let validated_staking_module = deps.api.addr_validate(&msg.staking_module_address)?;
+
+    STAKING_MODULE.save(deps.storage, &validated_staking_module)?;
```

Comment:
Can we do any additional validation here? This is fine, but would be nice to be able to do a proper check.

Reply:
Yeah I need to look into this, but initially my answer is no. The sort of thing here would be front end hard coded for the chain deployed on.